### ISSUE TITLE
Rule Request: unused control flow labels 

### ISSUE THREAD

**Issue by masters3d** [realm/SwiftLint#2227]
### Rule Request
>1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Typescript just added a way to identify labels which are not being used. I think this would be really nice to have in swift as lint rule. 

https://github.com/Microsoft/TypeScript/pull/24037/files

>2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```c
label2:  /// label2 not being used. Consider removing
    while (true) {}
```

do, while, repeat should be supported

non triggering examples:

```c
label2: 
    while (true) { break label2}
```
>3. Should the rule be configurable, if so what parameters should be configurable?

none

>4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

probably opt in
